File: TOKI-Learning/c++/bilangan_agak_prima.cpp
#include <cstdio>
#include <cmath>

int main()
{
    int n, faktor, i, j, bi;
    scanf("%d", &n);
    for (i = 0; i < n; i++)
    {
        faktor = 2;
        scanf("%d", &bi);
        for (j = 2; j <= trunc(sqrt(bi)); j++)
        {
            if (bi % j == 0)
                faktor += 2;
            if (faktor > 4)
                break;
        }
        if (faktor <= 4)
            printf("YA\n");
        else
            printf("BUKAN\n");
    }
    return 0;
}

File: TOKI-Learning/pascal/bilangan_agak_prima.pas
var
	n, i, angka, j, faktor : cardinal;
begin
	readln(n);
	for i := 1 to n do
	begin
		readln(angka);
		faktor := 0;
		for j := 2 to trunc(angka/3) do
			if faktor > 2 then
				break
			else if angka mod j = 0 then
				inc(faktor);
				
		if faktor <= 2 then
			writeln('YA')
		else
			writeln('BUKAN');
	end;
end.